src/rascal_utility.cc

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file   rascal_utility.cc
+ *
+ * Copyright  2019 COSMO (EPFL), LAMMM (EPFL)
+ *
+ * Rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * Rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include <cstdlib>
+#include <fstream>
+#include <iterator>
+
+#include "rascal_utility.hh"
+
+#ifdef __GNUG__
+#include <cxxabi.h>
+#endif
+
+std::string rascal::internal::type_name_demangled(const char * name) {
+#ifdef __GNUG__
+  int status = 0;
+  char * demangled = abi::__cxa_demangle(name, nullptr, nullptr, &status);
+  auto result = std::string(demangled);
```

Comment:
To follow the code convention strictly here one should use bracket. I personally feel bracket initialization is has no advantage when using auto so I would keep this as is.

Reply:
Actually there is a case in the coding convention stating exactly this:

> Never assign a braced-init-list to an auto local variable. In the single element case, what this means can be confusing.

So it should _not_ be with braces.

Reply:
Outside of this PR, but what is the rational for enforcing brace initialization as the coding style?

Reply:
If I remember correctly the idea is that when using the brace initializer, no narrowing is allowed. And to be consistent, we encourage it in the coding convention to use `{}` when possible `()` when necessary and try to avoid `=`.

Reply:
> If I remember correctly the idea is that when using the brace initializer, no narrowing is allowed.

Only when using a variable (see below) but OK.

```cpp
long long a{44};
short b{a};    // error
short c{44ll}; // ok
```

---

My main issue with it is that the syntax is a bit alien, especially for people coming from Python (which mean a lot of scientists). It also make the code a bit harder to parse when reading it, even more so if the initializer is long:

```
char * demangled = abi::__cxa_demangle(name, nullptr, nullptr, &status);
// VS
char * demangled{abi::__cxa_demangle(name, nullptr, nullptr, &status)};
```

The brace initialization looks more like a function call than a variable creation.

---

Anyway, as long as we can find a way to have it automatically enforced, I can deal with either. But I would rather not spend reviewer and implementer time checking for this =)